src/signTxOutput.c

```diff
@@ -468,11 +472,20 @@ static void signTxOutput_handleTokenAPDU(uint8_t* wireDataBuffer, size_t wireDat
 		read_view_t view = make_read_view(wireDataBuffer, wireDataBuffer + wireDataSize);
 
 		VALIDATE(view_remainingSize(&view) >= 4, ERR_INVALID_DATA);
-		token->assetNameSize = parse_u4be(&view);
-		VALIDATE(token->assetNameSize <= ASSET_NAME_SIZE_MAX, ERR_INVALID_DATA);
+		const size_t candidateAssetNameSize = parse_u4be(&view);
+
+		VALIDATE(candidateAssetNameSize <= ASSET_NAME_SIZE_MAX, ERR_INVALID_DATA);
+		ASSERT(candidateAssetNameSize <= SIZEOF(token->assetNameBytes));
+		//TODO not sure if the C99 dynamic array should be used on this project?
+		uint8_t candidateAssetNameBytes[ASSET_NAME_SIZE_MAX];
+
+		view_memmove(candidateAssetNameBytes, &view, candidateAssetNameSize);
+		VALIDATE(0 == subctx->currentToken || cbor_mapKeyFulfillsCanonicalOrdering(
+			token->assetNameBytes, token->assetNameSize,
+			candidateAssetNameBytes, candidateAssetNameSize) < 0, ERR_INVALID_DATA);
 
-		ASSERT(token->assetNameSize <= SIZEOF(token->assetNameBytes));
-		view_memmove(token->assetNameBytes, &view, token->assetNameSize);
+		token->assetNameSize = candidateAssetNameSize;
+		memcpy(token->assetNameBytes, candidateAssetNameBytes, candidateAssetNameSize);
```

Comment:
Before `memcpy/memmove`, always add an assertion that there is enough space in the target buffer.
(Several times in this PR.)

Reply:
Even view_memmove?